scenario/state.py

```diff
@@ -140,6 +140,83 @@ def copy(self) -> "Self":
         return copy.deepcopy(self)
 
 
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    auth_type: str
+    """Authentication type."""
+
+    attributes: Dict[str, str] = dataclasses.field(default_factory=dict)
+    """A dictionary containing cloud credentials.
+
+    For example, for AWS, it contains `access-key` and `secret-key`;
+    for Azure, `application-id`, `application-password` and `subscription-id`
+    can be found here.
+    """
+
+    redacted: List[str] = dataclasses.field(default_factory=list)
+    """A list of redacted secrets."""
+
+    @classmethod
+    def from_dict(cls, d: Dict[str, Any]) -> "CloudCredential":
+        """Create a new CloudCredential object from a dictionary."""
+        return cls(
+            auth_type=d["auth-type"],
+            attributes=d.get("attrs") or {},
+            redacted=d.get("redacted") or [],
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class CloudSpec:
+    type: str
+    """Type of the cloud."""
```

Comment:
It feels like this should be an enum, no?

Reply:
I disagree on this one. Juju doesn't do an enum (just a string), plus, cloud names are tied to company life cycles and come and go over time, and I don't think we want to be having to add to the enum when a new cloud is added. That said, if we can point to a simple place in Juju code for their official list, that might change my mind.

Reply:
Fair enough -- I could go either way. I suppose `scenario.CloudType.AWS` is not reasonable.

Reply:
Thanks for all the comments above. Let's keep it as a str for now, similar in OF.

Reply:
It seems strange to me to have AuthType as an enum but *not* cloud type. Let's do none or both. It looks like we're leaning towards enums, so let's do enums for both. Again, link to the Juju source in a comment.

Reply:
FYI: iirc we had a discussion in the sync and decided to keep them as they are.

Reply:
Sorry for getting back at it so late, I just finished some other tasks and could get back at it now.

I knew according to the discussion in the PR that it should be enum, but in one of our daily syncs we agreed to keep them as strs for now. I just confirmed this with Tony. Besides, see these two replies:

- https://github.com/canonical/ops-scenario/pull/119#issuecomment-2048846273
- https://github.com/canonical/ops-scenario/pull/119#issuecomment-2060217617

Closing this comment for now.